### PR TITLE
VFB-143 add parcel details link

### PR DESCRIPTION
**Pull request by mtharkins**
## What's changed
Links to client page and edit client page have been added as buttons to the modal of each parcel. 
## Screenshots / Videos
| Before     | After      |
|------------|------------|
| image | image |

## Checklist
- [x] The ticket is up-to-date - Please document any deviations from the original approach if there is any.
- [x] I have documented the testing steps for QA 
- [x] Make sure you've verified it works via `npm run dev`
- [x] Make sure you've verified it works via `npm run build` and `npm run start`
- [x] Make sure you've fixed all linting problems with `npm run lint_fix`
- [x] Make sure you've tested via `npm run test`

If you have made any changes to the database...
  - [x] The migration files are up-to-date with my final set up (`npx supabase db diff -f <name_of_migration>` should create nothing at this point)
  - [x] I have updated the typescript definitions for the database with `db:local:generate_types`
  - [x] I have modified the seed in `seed.mts` if appropriate
  - [x] If I have modified the seed, I have also generated the seed with `npm run db:generate_seed` 
  - [x] With my final set up, I can run `npm run dev:reset_supabase` without any errors.

### CODE REVIEW

**Comment by mtharkins** [comment]
Tests passed locally but failing here, re-checking to see what is causing the issue.

File: src/app/parcels/ParcelsPage.tsx
Comment (rickaw) [inline src/app/parcels/ParcelsPage.tsx:492]: Could you explain how `selectedPrimaryKey` is set when we directly go to that modal page `.../pacels/pacelId=...`? I don't think it's propertly set. To demonstrate this, tru
- opening the modal by clicking on a row
- copy the URL
- open that URL in a new tab
- click "See client"

File: src/app/parcels/ParcelsPage.tsx
Comment (mtharkins) [reply src/app/parcels/ParcelsPage.tsx:497]: Makes more sense, changed

File: src/app/parcels/ParcelsPage.tsx
Comment (mtharkins) [reply src/app/parcels/ParcelsPage.tsx:492]: I see what you mean now, I've amended the useEffect to set the primaryKey and I can access the client page now even if I  copy the URL to another tab.

File: src/app/parcels/ParcelsPage.tsx
Comment (mtharkins) [reply src/app/parcels/ParcelsPage.tsx:172]: Changed

File: src/app/parcels/ParcelsPage.tsx
Comment (mtharkins) [reply src/app/parcels/ParcelsPage.tsx:496]: Implemented

File: src/app/parcels/ParcelsPage.tsx
Comment (mtharkins) [reply src/app/parcels/ParcelsPage.tsx:201]: split into two separate useEffects

File: A_NCD2POTB_BCD.pdf
Comment (rickaw) [inline A_NCD2POTB_BCD.pdf:1]: unintended file?

File: src/app/clients/getExpandedClientDetails.ts
Comment (rickaw) [inline src/app/clients/getExpandedClientDetails.ts:124]: i think this change is covered in a different ticket (153)

File: src/app/parcels/ParcelsPage.tsx
Comment (rickaw) [inline src/app/parcels/ParcelsPage.tsx:194]: As discussed, i think we'd want to fetch from supabase to the parcel data, and then the client id from there.

File: A_NCD2POTB_BCD.pdf
Comment (mtharkins) [reply A_NCD2POTB_BCD.pdf:1]: Removed

File: src/app/parcels/ParcelsPage.tsx
Comment (mtharkins) [reply src/app/parcels/ParcelsPage.tsx]: Added

**Review by rickaw — APPROVED** [review]

**Review by rickaw — CHANGES_REQUESTED** [review]
sorry approved the wrong PR loool

File: src/app/parcels/ParcelsPage.tsx
Comment (rickaw) [inline src/app/parcels/ParcelsPage.tsx:195]:
```suggestion
        const fetchAndSetClientIdForSelectedParcel = async (): Promise<void> => {
```

File: src/app/parcels/ParcelsPage.tsx
Comment (rickaw) [inline src/app/parcels/ParcelsPage.tsx:202]: Can we use logError please? Should we also throw a DatabaseError?
```suggestion
                void logError("...")
```

File: src/app/parcels/ParcelsPage.tsx
Comment (rickaw) [inline src/app/parcels/ParcelsPage.tsx:206]: I think typescirpt knows data exists since you've caught the case where error exists
```suggestion
            if (data.length > 0) {
```

**Review by rickaw — CHANGES_REQUESTED** [review]

File: src/app/parcels/ParcelsPage.tsx
Comment (mtharkins) [reply src/app/parcels/ParcelsPage.tsx:365]: Changed

File: src/app/parcels/ParcelsPage.tsx
Comment (mtharkins) [reply src/app/parcels/ParcelsPage.tsx:372]: Changed

File: src/app/parcels/ParcelsPage.tsx
Comment (mtharkins) [reply src/app/parcels/ParcelsPage.tsx:376]: Changed

File: src/app/parcels/ParcelsPage.tsx
Comment (mtharkins) [reply src/app/parcels/ParcelsPage.tsx]: Changed

File: src/app/parcels/ParcelsPage.tsx
Comment (mtharkins) [reply src/app/parcels/ParcelsPage.tsx]: Changed

File: src/app/parcels/ParcelsPage.tsx
Comment (mtharkins) [reply src/app/parcels/ParcelsPage.tsx]: Changed and removed redundant code

File: computer_names.csv
Comment (rickaw) [inline computer_names.csv:1]: unintented file

File: src/app/parcels/ParcelsPage.tsx
Comment (rickaw) [inline src/app/parcels/ParcelsPage.tsx:371]: Sorry I wasn't clear before, let's log something useful in `logError` (which might be called something else now if i remember correctly)

Maybe some error message, and also pass the `error` itself?

File: src/app/parcels/ParcelsPage.tsx
Comment (rickaw) [inline src/app/parcels/ParcelsPage.tsx:376]: I don't think this is indented correctly?

File: src/app/parcels/ParcelsPage.tsx
Comment (rickaw) [inline src/app/parcels/ParcelsPage.tsx:735]: i don't think the indentation is right - try linting

File: computer_names.csv
Comment (mtharkins) [reply computer_names.csv:1]: Sorry I thought I had stopped this from happening, removed

File: src/app/parcels/ParcelsPage.tsx
Comment (mtharkins) [reply src/app/parcels/ParcelsPage.tsx:371]: Apologies, I copied your suggestion as a placeholder and forgot to replace the ellipsis. Added error text as well as the message.

File: src/app/parcels/ParcelsPage.tsx
Comment (mtharkins) [reply src/app/parcels/ParcelsPage.tsx:376]: Amended

File: src/app/parcels/ParcelsPage.tsx
Comment (mtharkins) [reply src/app/parcels/ParcelsPage.tsx:735]: I've rerun lint_fix which seems to have updated this correctly now

File: src/app/parcels/ParcelsPage.tsx
Comment (rickaw) [inline src/app/parcels/ParcelsPage.tsx:371]: You could do this do log `{ error }` as metadata. (The second argument logError is metadata) 
NB `{ error }` is equivalent to `{ error: error }`
```suggestion
                void logError(`Error fetching clientId from database`, { error });
```

Could we rebase or merge dev to your branch please? I think `logError` is called something else now so you'll probs need to rename

File: src/app/parcels/ParcelsPage.tsx
Comment (rickaw) [inline src/app/parcels/ParcelsPage.tsx:379]: To explicitly indicate that the promise is ignored
```suggestion
        void fetchAndSetClientIdForSelectedParcel();
```

**Review by rickaw — APPROVED** [review]
LGTM - feel free to merge once you've rebased / merged dev

File: src/app/parcels/ParcelsPage.tsx
Comment (mtharkins) [reply src/app/parcels/ParcelsPage.tsx:371]: Changed and merged

File: src/app/parcels/ParcelsPage.tsx
Comment (mtharkins) [reply src/app/parcels/ParcelsPage.tsx:379]: Added

**Pull request merged**
